packages/@lwc/wire-service/src/index.ts

```diff
@@ -14,20 +16,31 @@ export function register(
     adapterEventTargetCallback: (eventTarget: WireEventTarget) => void
 ) {
     if (adapterId == null || !(adapterId instanceof Object)) {
-        new TypeError('adapter id must be an object or a function');
+        throw new TypeError('adapter id must be an object or a function');
     }
     if (typeof adapterEventTargetCallback !== 'function') {
-        new TypeError('adapter factory must be a callable');
+        throw new TypeError('adapter factory must be a callable');
     }
     if ('adapter' in adapterId) {
-        new TypeError('adapter id is already associated to an adapter factory');
+        throw new TypeError('adapter id is already associated to an adapter factory');
     }
-    adapterId.adapter = class extends WireAdapter {
-        constructor(dataCallback: dataCallback) {
-            super(dataCallback);
-            adapterEventTargetCallback(this.eventTarget);
+
+    const adapterClass = freeze(
```

Comment:
you should probably freeze it on the next line after declaration, and make `AdapterClass` an actual class to it retains the proper name instead of an anonymous class

Reply:
makes sense, fixing it.